src/components/dialogs/AddToSocialGroupDialog.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useCallback, useState } from 'react';
+import { get } from 'lodash-es';
+import { FormattedMessage } from 'react-intl';
+
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import Select from '@material-ui/core/Select';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+
+import useSiteSettings from '../../models/site/useSiteSettings';
+import useSocialGroup from '../../models/socialGroups/useSocialGroup';
+import useSocialGroups from '../../models/socialGroups/useSocialGroups';
+import usePatchSocialGroup from '../../models/socialGroups/usePatchSocialGroup';
+import CustomAlert from '../Alert';
+import InputRow from '../fields/edit/InputRow';
+import Button from '../Button';
+import StandardDialog from '../StandardDialog';
+
+export default function AddToSocialGroupDialog({
+  open,
+  onClose,
+  individualGuid,
+}) {
+  const [selectedGroup, setSelectedGroup] = useState('');
+  const [selectedRole, setSelectedRole] = useState('');
+
+  const { data: socialGroups, loading: socialGroupsLoading } =
+    useSocialGroups();
+  const { data: siteSettings, loading: siteSettingsLoading } =
+    useSiteSettings();
+  const {
+    data: selectedGroupData,
+    loading: selectedGroupDataLoading,
+  } = useSocialGroup(selectedGroup);
```

Comment:
These need some sort of error handling that lets the user know if the form is unusable.

Reply:
added